mvc/models/model-binding.rst

```diff
@@ -1,8 +1,89 @@
+
+In order for binding to happen, members of complex model types must be public, writable properties containing a default public constructor. Public fields are not bound. If properties are arrays or IEnumerable types, the array properties or types in the IEnumerable must be writable. Model binding handles some objects differently than most. ``ICollection<T>`` is supported for read only properties.  
+
+When a parameter is bound, model binding stops looking for parameters with that name and it moves on to bind the next parameter. If binding fails, MVC does not throw an error. The error is recorded in the ``ModelState.Errors`` collection, and model binding attempts to bind to the next parameter or property. You can query for model state errors by checking the ``ModelState.IsValid`` method. 
+
+.. Note:: The ``ModelState.Errors`` collection is a property of the controller's ``ModelStateDictionary`` object, containing name-value pairs. It's rarely necessary to query this collection yourself; if there are items in this collection, then ``ModelState.IsValid`` returns ``false``. 
+
+Additionally, there are some special data types that MVC must consider when performing model binding:
+
+- ``IFormFile``: Uploaded files that are part of the HTTP request.
+- ``IFormCollection``: The collection of key-value pairs that represent the form data.
+- ``CancelationToken``: Used to cancel activity in asynchronous controllers.
+
+Once model binding is complete, `validation <https://docs.asp.net/projects/mvc/en/latest/models/validation.html>`_ occurs. Default model binding works great for the vast majority of development scenarios. It is also extensible so if you have unique needs you can customize the built-in behavior.  
+
+Customize model binding behavior with attributes 
+--------------------------------------------------------
+MVC contains several attributes that you can use to direct its default model binding behavior to a different source. For example, you can specify whether binding is required for a field, or if it should never happen at all by using the ``[BindRequired]`` or ``[BindNever]`` attributes. Alternatively, you can override the default value, and specify the model binder's data source. Below is a list of model binding attributes:
```

Comment:
> field

property

> you can override the default value

I interpreted this to mean a default value for the parameter (e.g. `Get(int id = -23)`) when I first read it. Suggest changing this to "default data source".